docs/design/maven.md

```diff
@@ -0,0 +1,130 @@
+# Generic artifact fetching
+
+## Introduction
+
+This document will describe high-level implementation overview for supporting fetching maven artifacts in cachi2. This
+proposal depends on generic artifact fetching, which is described in this [design PR][generic-pr].
+
+## Context
+
+[Maven] is a Java ecosystem build system and functions basically as a package manager, among other things. Maven provides a
+standardized way to specify dependencies in a `pom.xml` file, which is then used to fetch dependencies from a maven repository.
+This feature in cachi2 isn't concerned with pre-fetching those, but rather to provide a way to fetch individual artifacts
+from maven repository. This is useful for cases where the build only needs specific artifacts from big projects, saving
+time and bandwidth.
+
+### Maven artifacts
+
+[Maven artifacts][maven-artifacts] are defined as anything that resides in a maven repository. Maven has a special way of
+addressing artifacts in its repositories called coordinates, which is usually represented by `GAV` (Group, Artifact, Version)
```

Comment:
```suggestion
addressing artifacts in its repositories called coordinates, which are usually represented by `GAV` tuple (`GAV` standing for 'Group, Atifact, Version').
```
(if I read the paragraph correctly). Are Group and Artifact names? From the paragraph below it might seem that this tuple must also contain 'type' and 'classifier' fields, are those missing?

Reply:
Adjusted the wording. This is a maven quirk, while there's more to it than just the three, the name stuck. [Docs ref](https://maven.apache.org/repositories/artifacts.html)

> Artifact coordinates, are most often represented as groupId:artifactId:version, or GAV in short or when informally used (please note that Artifact coordinates has more fields, but for brevity we still call the coordinates “GAV”, not “GAVCE”).